che--typescript/src/internal/action/impl/list-workspaces-action.ts

```diff
@@ -64,7 +66,18 @@ export class ListWorkspacesAction {
                     // Create Ascii array
                     let rows : Array<Array<string>> = new Array<Array<string>>();
                     workspaceDtos.forEach((workspaceDto : any) => {
-                        rows.push([workspaceDto.getConfig().getName(), workspaceDto.getId(), workspaceDto.getStatus()]);
+                        let syncAgent : string = "";
+                        if (this.syncAgentCheck === true){
+                            let defaultEnv:string = workspaceDto.getConfig().getDefaultEnv();
+                            let agents:Array<string> = workspaceDto.getConfig().getEnvironments().get(defaultEnv).getMachines().get("dev-machine").getAgents();
+                            if (agents.indexOf('org.eclipse.che.unison') !== -1) {
+                                syncAgent = "(Sync Agent)";
+                            }
+                            else{
+                                syncAgent = "(No Sync Agent)";
+                            }
+                        }
+                        rows.push([workspaceDto.getConfig().getName()+syncAgent, workspaceDto.getId(), workspaceDto.getStatus()]);
```

Comment:
it looks like a hack to display on list of workspace only the sync agent. What about other agents ? list-workspaces is not only for usage in File Sync 
Also it looks strange as agents are per machine so displaying them per workspace is not correct.